app/views/NotificationPrefView/index.js

```diff
@@ -0,0 +1,401 @@
+			value: 'default'
+		},
+		{
+			label: 'All Messages',
+			value: 'all'
+		},
+		{
+			label: 'Mentions',
+			value: 'mentions'
+		},
+		{
+			label: 'Nothing',
+			value: 'nothing'
+		}
+	],
+	audioNotifications: [
+		{
+			label: 'Default',
+			value: 'default'
+		},
+		{
+			label: 'All Messages',
+			value: 'all'
+		},
+		{
+			label: 'Mentions',
+			value: 'mentions'
+		},
+		{
+			label: 'Nothing',
+			value: 'nothing'
+		}
+	],
+	mobilePushNotifications: [
+		{
+			label: 'Default',
+			value: 'default'
+		},
+		{
+			label: 'All Messages',
+			value: 'all'
+		},
+		{
+			label: 'Mentions',
+			value: 'mentions'
+		},
+		{
+			label: 'Nothing',
+			value: 'nothing'
+		}
+	],
+	emailNotifications: [
+		{
+			label: 'Default',
+			value: 'default'
+		},
+		{
+			label: 'All Messages',
+			value: 'all'
+		},
+		{
+			label: 'Mentions',
+			value: 'mentions'
+		},
+		{
+			label: 'Nothing',
+			value: 'nothing'
+		}
+	],
+	desktopNotificationDuration: [
+		{
+			label: 'Default',
+			value: 0
+		},
+		{
+			label: '1 second',
+			value: 1
+		},
+		{
+			label: '2 seconds',
+			value: 2
+		},
+		{
+			label: '3 seconds',
+			value: 3
+		},
+		{
+			label: '4 seconds',
+			value: 4
+		},
+		{
+			label: '5 seconds',
+			value: 5
+		}
+	],
+	audioNotificationValue: [
+		{
+			label: 'None',
+			value: 'none None'
+		},
+		{
+			label: 'Default',
+			value: '0 Default'
+		},
+		{
+			label: 'Beep',
+			value: 'beep Beep'
+		},
+		{
+			label: 'Ding',
+			value: 'ding Ding'
+		},
+		{
+			label: 'Chelle',
+			value: 'chelle Chelle'
+		},
+		{
+			label: 'Droplet',
+			value: 'droplet Droplet'
+		},
+		{
+			label: 'Highbell',
+			value: 'highbell Highbell'
+		},
+		{
+			label: 'Seasons',
+			value: 'seasons Seasons'
+		}
+	]
+
+};
+
+@connect(state => ({
+	server: state.server
+}))
+export default class NotificationPrefView extends React.Component {
+	static navigationOptions = () => ({
+		title: I18n.t('Notification_Preference')
+	})
+
+	static propTypes = {
+		navigation: PropTypes.object
+	}
+
+	constructor(props) {
+		super(props);
+		this.rid = props.navigation.getParam('rid');
+		this.rooms = database.objects('subscriptions').filtered('rid = $0', this.rid);
+		this.room = JSON.parse(JSON.stringify(this.rooms[0] || {}));
+		this.state = {
+			room: JSON.parse(JSON.stringify(this.rooms[0] || {}))
+		};
+	}
+
+	renderSwitch = (key) => {
+		const { room } = this.state;
+		return (
+			<Switch
+				value={!room[key]}
+				testID={key}
+				trackColor={SWITCH_TRACK_COLOR}
+				onValueChange={value => this.onValueChange(key, !value)}
+			/>
+		);
+	}
+
+	onValueChange = (key, value) => {
+		const { room: newRoom } = this.state;
+		newRoom[key] = value;
+		this.setState({ room: newRoom });
+	}
+
+	renderPicker = (key) => {
+		const { room } = this.state;
+		return (
+			<RNPickerSelect
+				testID={key}
+				value={room[key]}
+				textInputProps={{ style: styles.pickerText }}
+				useNativeAndroidPickerStyle={false}
+				placeholder={{}}
```

Comment:
Is it necessary?

Reply:
Yes else it will have a default placeholder that says something and it comes up in options too